src/optimize_module.cpp

```diff
@@ -36,8 +36,12 @@ void optimize_module::apply(module_pass_manager& mpm) const
 {
     for(int i = 0; i < 2; i++)
     {
-        mpm.run_pass(simplify_reshapes{});
-        mpm.run_pass(simplify_algebra{});
+        // loop to further optimize after initial transformations
```

Comment:
can't we just increase the outer loop trip count ?

Reply:
no because `propagate_constant` will otherwise compute the unoptimized shape for the literal.

Reply:
A better solution would be to update constant propagation to not fold when the inputs are broadcasted across a common axis, but that will be more work than just simply rerunning the passes. So we can save that for another PR.

Reply:
I've added a basic `optimize_module` test to combine some of the transformations. I don't know the full steps in `simplify_algebra` that led to the broadcast+transpose scalar in the bert model, and it might take me more time to figure out. Maybe I can add that later in a separate PR?